modules/stochastic_tools/src/actions/ParameterStudyAction.C

```diff
@@ -0,0 +1,711 @@
+registerMooseAction("StochasticToolsApp", ParameterStudyAction, "add_multi_app");
+registerMooseAction("StochasticToolsApp", ParameterStudyAction, "add_transfer");
+registerMooseAction("StochasticToolsApp", ParameterStudyAction, "add_output");
+registerMooseAction("StochasticToolsApp", ParameterStudyAction, "add_vector_postprocessor");
+registerMooseAction("StochasticToolsApp", ParameterStudyAction, "add_reporter");
+registerMooseAction("StochasticToolsApp", ParameterStudyAction, "add_control");
+
+InputParameters
+ParameterStudyAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Builds objects to set up a basic parameter study.");
+
+  // Parameters to define what we are studying
+  params.addRequiredParam<FileName>(
+      "input", "The input file containing the physics for the parameter study.");
+  params.addRequiredParam<std::vector<std::string>>(
+      "parameters", "List of parameters being perturbed for the study.");
+  params.addParam<std::vector<ReporterName>>(
+      "quantities_of_interest",
+      "List of the reporter names (object_name/value_name) "
+      "that represent the quantities of interest for the study.");
+  params.addParam<bool>("compute_statistics",
+                        true,
+                        "Whether or not to compute statistics on the 'quantities_of_interest'. "
+                        "This will compute mean and standard deviation with 0.01, 0.05, 0.1, 0.9, "
+                        "0.95, and 0.99 confidence intervals.");
+
+  // Parameters for the sampling scheme
+  params.addRequiredParam<MooseEnum>(
+      "sampling_type", samplingTypes(), "The type of sampling to use for the parameter study.");
+  params.addParam<unsigned int>("seed", 0, "Random number generator initial seed");
+
+  // Parameters for multi app
+  MooseEnum modes(
+      "normal=0 batch-reset=1 batch-restore=2 batch-keep-solution=3 batch-no-restore=4");
+  params.addParam<MooseEnum>(
+      "multiapp_mode",
+      modes,
+      "The operation mode, 'normal' creates one sub-application for each sample."
+      "'batch' creates one sub-app for each processor and re-executes for each local sample. "
+      "'reset' re-initializes the sub-app for every sample in the batch. "
+      "'restore' does not re-initialize and instead restores to first sample's initialization. "
+      "'keep-solution' re-uses the solution obtained from the first sample in the batch. "
+      "'no-restore' does not restore the sub-app."
+      "The default will be inferred based on the study.");
+  params.addParam<unsigned int>(
+      "min_procs_per_sample",
+      1,
+      "Minimum number of processors to give to each sample. Useful for larger, distributed mesh "
+      "solves where there are memory constraints.");
+
+  // Samplers ///////////////////////////
+  // Parameters for Monte Carlo and LHS
+  params.addParam<dof_id_type>(
```

Comment:
or maybe just do `params += Sampler::validParams()` (not sure whether it will work or not).

Reply:
I struggled with this a bit. I concluded that I don't want all the parameters from the samplers and distributions in this action's parameters. For the samplers, many of the parameters are required (`num_rows` and `distributions` as examples) which would not be relevant for other sampling types and it would be just as difficult to mark them as not required as it would be to just add them independently here. For the distributions, many of the distributions have the same parameter names, so determining which distribution it goes to is impossible I think. The samplers and distributions are pretty static, so I don't see too much problem with the duplication.